packages/components/app/styles/components/table.scss

```diff
@@ -19,175 +19,166 @@ $hds-table-cell-padding-medium: 14px 16px 13px 16px; // the 1px difference is to
+
+      // border between two cells (we emulate a cell border slightly detached from the top/bottom borders)
+
+      + th::before {
+        position: absolute;
+        top: 6px;
+        bottom: 6px;
+        left: -1px; // we need to offset the border by 1px to render a right border of the previous cell
+        width: 1px;
+        background-color: $hds-table-border-color;
+        content: "";
+        pointer-events: none;
       }
     }
-  }
 
-  .hds-table__th,
-  .hds-table__th-sort {
-    min-height: $hds-table-header-height;
-  }
-}
+    // horizontal alignment
 
-.hds-table__th,
-.hds-table__th-sort {
-  text-align: left;
-  border-top: none;
-  border-right: none;
-  border-bottom: $hds-table-border-width solid $hds-table-border-color;
-  border-left: none;
-}
+    .hds-table__th--align-center,
+    .hds-table__td--align-center {
+      text-align: center;
 
-.hds-table__th {
-  padding: $hds-table-cell-padding-medium;
-}
+      .hds-table__th-content {
+        justify-content: center;
+      }
+    }
 
-.hds-table__th-sort {
-  padding: 0;
+    .hds-table__th--align-right,
+    .hds-table__td--align-right {
+      text-align: right;
 
-  button {
-    width: 100%;
-    height: 100%;
-    min-height: $hds-table-header-height;
-    margin: 0;
-    padding: $hds-table-cell-padding-medium;
-    text-align: inherit;
-    background-color: transparent;
-    border: 1px solid transparent;
-    border-radius: inherit;
-
-    .hds-table__th-sort--button-content {
-      display: flex; // we can't put flex on the button itself so we drop the content into its own container
-      align-items: center;
-
-      .flight-icon {
-        flex: none;
-        margin-left: 8px;
-        color: var(--token-color-foreground-action);
+      .hds-table__th-content {
+        justify-content: flex-end;
       }
     }
 
-    &:hover,
-    &.mock-hover {
-      color: var(--token-color-foreground-strong);
-      background-color: var(--token-color-palette-neutral-200);
-      cursor: pointer;
-    }
+    // border radius: target first and last th elements in the row
 
-    @include hds-focus-ring-with-pseudo-element($radius: inherit);
+    &:first-of-type {
+      th:first-child {
+        border-top-left-radius: $hds-table-inner-border-radius;
+      }
 
-    &:active,
-    &.mock-active {
-      color: var(--token-color-foreground-strong);
-      background-color: var(--token-color-palette-neutral-300);
+      // last th in the row, border radius
+      th:last-child {
+        border-top-right-radius: $hds-table-inner-border-radius;
+      }
     }
   }
 }
 
-
-// TABLE BODY
-
-// TABLE STRIPING ROWS
-.hds-table--striped {
-  .hds-table__tbody {
-    .hds-table__tr:nth-child(even) {
-      background-color: var(--token-color-surface-faint);
-    }
-  }
+.hds-table__th-content {
+  display: flex;
+  gap: 8px;
+  align-items: center;
+  justify-content: flex-start;
 }
 
-// TABLE DENSITY (TBODY ROW HEIGHT)
+.hds-table__th-button {
```

Comment:
not too familiar with the style of this button, but wanted to check if we considered using our button mixins

Reply:
@alex-ju good thinking! I tried but there are some differences, so I would have to write a bunch of overrides, and at that point  I am not sure it would be worth it: what do you think?
<img width="728" alt="screenshot_3366" src="https://github.com/hashicorp/design-system/assets/686239/fd44a13e-4cea-46e2-9a81-d40fce301e1a">

Reply:
I just compared the definitions in the mixin and the ones for this button and there's not as much of an overlap. we can always use other opportunities to do such refactoring with no impact on consumers